src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -376,6 +376,16 @@ private MsalAccessTokenCacheItem FindAccessTokenCommon
                             .Where(item => item.HomeAccountId.Equals(requestParams.Account?.HomeAccountId.Identifier, StringComparison.OrdinalIgnoreCase))
                             .ToList();
                 }
+
+                if (tokenCacheItems.Count > 1)
```

Comment:
I think this needs more explanations. TokenCache storage is complex and once you understand how it works, it pays to add a few more comments for everybody (including yourself in the future). 

So you're saying that if there are more than 1 results (when would this happen?), filter the one that uses the canonical authority (why?). 

So you could actually filter all items out?

Reply:
I'm going to send an email w/more details. And included more in the [issue](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/695) thanks for all the great questions.

---
In reply to: [239032166](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/708#discussion_r239032166) [](ancestors = 239032166)

Reply:
Perhaps simply make a method which says filter to authority specified by the caller... or something like that. I would say you don't have to call it canonical, as that is likely not what is central to understanding what is going on.

---
In reply to: [239286789](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/708#discussion_r239286789) [](ancestors = 239286789,239032166)

Reply:
good ideas..thank you. done.

---
In reply to: [239342110](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/708#discussion_r239342110) [](ancestors = 239342110,239286789,239032166)